addons/delivery/wizard/choose_delivery_carrier_views.xml

```diff
@@ -38,7 +38,10 @@
                 </div>
                 <footer>
                     <button name="button_confirm" invisible="not context.get('carrier_recompute')" type="object" string="Update" class="btn-primary" data-hotkey="q"/>
-                    <button name="button_confirm" invisible="context.get('carrier_recompute')" type="object" string="Add" class="btn-primary" data-hotkey="q"/>
+                    <button name="button_confirm" invisible="context.get('carrier_recompute') or not display_price" type="object" string="Add" class="btn-primary" data-hotkey="q"/>
+                    <button name="button_confirm" invisible="context.get('carrier_recompute') or display_price"
```

Comment:
If you indent xml nodes, please use one attribute for each line:

```xml
    <button
        name=
        type=
        invisible=
```

Also, could you make sure the confirmation message is not shown when the carrier is 'fixed' or 'based_on_rule', even if the price is zero?  It only makes sense to display the warning for carriers where the rates have to be fetched through the button.